kpatch-build/create-diff-object.c

```diff
@@ -3402,30 +3440,69 @@ static void kpatch_create_callbacks_objname_rela(struct kpatch_elf *kelf, char *
 }
 
 /*
+ * Allocate the mcount/patchable_function_entry sections which must be done
+ * before the patched object is torn down so that the section flags can be
+ * copied.
+ */
+static void kpatch_alloc_mcount_sections(struct kpatch_elf *kelf, struct kpatch_elf *kelfout)
+{
+	int nr;
+	struct symbol *sym;
+
+	nr = 0;
+	list_for_each_entry(sym, &kelfout->symbols, list)
+		if (sym->type == STT_FUNC && sym->status != SAME &&
+		    sym->has_func_profiling)
+			nr++;
+
+	/* create text/rela section pair */
+#ifdef __aarch64__
+{
+	struct section *sec, *tmp;
+
+	sec = create_section_pair(kelfout, "__patchable_function_entries", sizeof(void*), nr);
+
+	/*
+	 * Depending on the compiler the __patchable_function_entries section
+	 * can be ordered or not, copy this flag to the section we created to
+	 * avoid:
+	 * ld: __patchable_function_entries has both ordered [...] and unordered [...] sections
+	 */
+	tmp = find_section_by_name(&kelf->sections, "__patchable_function_entries");
+	sec->sh.sh_flags |= (tmp->sh.sh_flags & SHF_LINK_ORDER);
+	sec->sh.sh_flags = 1;
```

Comment:
sec->sh.sh_link = 1 ?

Reply:
Good catch, this resolves the issue @joe-lawrence reported above. I also thought we decided to copy it over from patched objectfile instead of always setting it.